### DISCUSSION TITLE
GraphQL mutations that fail RLS don't receive an error

### DISCUSSION THREAD

**Discussion by ky1ejs** [supabase/supabase#15325] [Questions] [answered]
This post continues on from discussion #13618.

When you make a GraphQL mutation that is unauthorized by RLS, no error is returned by Supabase. This is misleading when debugging.

![Screenshot 2023-06-23 at 18 00 40](https://github.com/supabase/supabase/assets/3501212/2fdc7374-813c-4c8c-982c-ed427f9e42b8)

I know [GraphQL is famous for responding 200 despite there being an error in the response payload](https://preview.redd.it/relo6xpykds71.jpg?auto=webp&s=595420ec296084b6b11ba65b9a3708163016c2c6), but Supabase doesn't even seem to be doing that.

1. Is this intentional and if so why?
2. If no, could Supabase return a more suitable response, describing that the request is unauthorized for this scenario?

**Comment by GaryAustin1** [comment, marked as answer]
Postgres does not return errors on RLS unless doing an insert.   It is by design for security reasons RLS will just hide the rows like they are not there.

**Reply by ky1ejs** [reply]
Got it, that makes sense! 

Also found a couple of StackOverflow answers that speak to this: 
* https://stackoverflow.com/questions/58521638/postgresql-row-level-security-does-not-throw-errors
* https://dba.stackexchange.com/questions/300776/postgres-rls-how-to-error-instead-of-ignoring-rows-not-matched-by-the-policy
